lib/Runtime/Base/ThreadContext.cpp

```diff
@@ -1083,7 +1087,7 @@ ThreadContext::AddPropertyRecordInternal(const Js::PropertyRecord * propertyReco
 #if DBG
     // Only Assert we can't find the property if we are not adding a symbol.
     // For a symbol, the propertyName is not used and may collide with something in the map already.
-    if (!propertyRecord->IsSymbol())
+    if (propertyNameLength > 0 && !propertyRecord->IsSymbol())
```

Comment:
Probably don't need this condition anymore? If we're adding a property record, the old invariant that it shouldn't already be in the map, should still hold

Reply:
It's for Debug to check `at this point, propertyRecord is not for a symbol`